tidy3d/components/material/multi_physics.py

```
@@ -98,6 +98,16 @@ class MultiPhysicsMedium(Tidy3dBaseModel):
         None, title="Charge properties", description="Specifies properties for Charge simulations."
     )
 
+    def __getattr__(self, name: str):
+        """Delegates calls to missing attributes to the inner media."""
+        if self.optical is not None and hasattr(self.optical, name):
```

I just noticed that this precedence order of `optical`, `heat` and `charge` might be sub-optimal and we should check whether there is a conflict between the media. For example the optical permittivity might be different than the one of the charge medium. Should we raise an error in this case?

---

Yeah so as we just chatted, ultimately this PR will discover the issues that were inherited or coupled between solvers. So this generic delegation might cause errors or different behaviour users might not expect. So as we chatted maybe to have it per indivdual fucntions to reduce the scope.

---

For my understanding what is you guys' thinking now? Also should we get this in today for the testing of the next patch or better leave it for 2.9?

---

So the `MultiPhysicsMedium` is used in places where the rest of the code expects a subclass of the `AbstractMedium`. Thus, its missing some functions and properties that are expected, such as the `is_pec` property or `_eps_plot` function.
The current implementation of the fix delegates all calls to missing properties/functions to one of the respective inner media. This could backfire if two properties/functions are called the same but have different semantics. I feared that this might be the case, so I wanted to build in a check. But this still is not fail-safe as per discussion with @daquinteroflex .
Thus, I will change this implementation to be restricted to a fixed  set of properties that are delegated to their respective medium. I will add a note that this set might need to be extended in the future.

---

Got it, indeed that makes sense to me!